### PR TITLE
apply rules for [ to [[ and its closing counterpair

### LINKED ISSUE #1029: Inconsistency of indention for [[
Is this expected ?

``` r
# as I expect
styler::style_text("a(\nb)")
#> a(
#>   b
#> )
styler::style_text("a[\nb]")
#> a[
#>   b
#> ]

# inconsistent, no indent, no pushing ]] to next line
styler::style_text("a[[\nb]]")
#> a[[
#> b]]
```

<sup>Created on 2022-10-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by lorenzwalthert**
Closes #1029.

### CODE REVIEW

File: R/indent.R [hunk 1]
```diff
@@ -105,7 +112,12 @@ compute_indent_indices <- function(pd,
   if (is.null(token_closing)) {
     stop <- npd
   } else {
-    stop <- last(which(pd$token %in% token_closing)[needs_indention]) - 1
+    offset <- if (any(pd$token == "LBB")) {
+      2
```
Comment (IndrajeetPatil) [inline R/indent.R:116]:
```suggestion
      2L
```

File: R/indent.R [hunk 2]
```diff
@@ -105,7 +112,12 @@ compute_indent_indices <- function(pd,
...
+    } else {
+      1
```
Comment (IndrajeetPatil) [inline R/indent.R:118]:
```suggestion
      1L
```

File: R/indent.R [hunk 3]
```diff
@@ -105,7 +112,12 @@ compute_indent_indices <- function(pd,
   if (is.null(token_closing)) {
     stop <- npd
   } else {
-    stop <- last(which(pd$token %in% token_closing)[needs_indention]) - 1
+    offset <- if (any(pd$token == "LBB")) {
```
Comment (IndrajeetPatil) [inline R/indent.R:115]:
```suggestion
    offset <- if (any(pd$token == "LBB")) 2L else 1L
```

Comment (IndrajeetPatil) [inline R/indent.R:115]:
```suggestion
    offset <- if (any(pd$token == "LBB")) 2L else 1L
```

File: R/rules-line-breaks.R [hunk 4]
```diff
@@ -128,7 +128,7 @@ set_line_break_around_comma_and_or <- function(pd, strict) {
     (pd$token %in% ops) &
       (pd$lag_newlines > 0L) &
       (pd$token_before != "COMMENT") &
-      (lag(pd$token) != "'['")
+      !(lag(pd$token) %in% c("LBB", "'['"))
```
Comment (IndrajeetPatil) [inline R/rules-line-breaks.R:131]: Let's assign `c("LBB", "'['")` to `left_token` variable and reuse that variable?

Comment (IndrajeetPatil) [reply R/rules-line-breaks.R:131]: Because we are using this vector in more than one location.

**Review by IndrajeetPatil — APPROVED** [review]

**Review by IndrajeetPatil — APPROVED** [review]

File: R/rules-line-breaks.R (hunk 4 above)
Comment (lorenzwalthert) [reply R/rules-line-breaks.R]: I see, but I'd do that in another place. Will force push...

Comment (lorenzwalthert) [reply R/rules-line-breaks.R]: We can re-use it in another file and we have defined groups of tokens before top-level.

**Pull request merged**